lib/include/ert/nexus/nexus_plot.hpp

```diff
@@ -39,25 +44,266 @@ struct unexpected_eof : public std::runtime_error {
+    }
+    bool time( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.time == rhs.time;
+    }
+    bool max_perfs( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.max_perfs == rhs.max_perfs;
+    }
+    bool classname( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.classname == rhs.classname;
+    }
+    bool instancename( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.instancename == rhs.instancename;
+    }
+    bool varname( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.varname == rhs.varname;
+    }
+    bool value( const NexusData& lhs, const NexusData& rhs ) {
+        return lhs.value == rhs.value;
+    }
+}
+
+struct NexusPlot {
+    NexusHeader header;
+    std::vector< NexusData > data;
+
+    template< typename T >
+    std::vector< T > get_unique( T (&f)(const NexusData&) ) const {
+        std::vector< T > vec;
+        vec.reserve( this->data.size() );
+        std::transform( this->data.begin(), this->data.end(),
+                        std::back_inserter( vec ),
+                        f);
+        auto last = std::unique(vec.begin(), vec.end());
+        std::sort(vec.begin(), last);
+        last = std::unique(vec.begin(), last);
+        vec.erase(last, vec.end());
+        return vec;
+    }
+
+    std::vector< std::string > varnames(char (&classname)[9]) const {
+        std::array< char, 8> cn {{ ' ' }};
+        std::copy( std::begin(classname), std::end(classname),
+                   std::begin( cn ) );
+        std::replace(cn.begin(), cn.end(), '\0', ' ');
+        return this->varnames( cn );
+    }
+    std::vector< std::string > varnames(std::string classname) const {
+        std::array< char, 8> cn {{ ' ' }};
+        if (classname.size() > cn.max_size())
+            throw std::runtime_error("Could not get varnames, given classname was larger than 8 bytes");
+        std::copy( std::begin(classname), std::end(classname),
+                   std::begin( cn ) );
+        std::replace(cn.begin(), cn.end(), '\0', ' ');
+        return this->varnames( cn );
+    }
+    std::vector< std::string > varnames(std::array< char, 8 > classname) const {
+        std::vector< NexusData > of_class;
+        std::copy_if( this->data.begin(), this->data.end(),
+                      std::back_inserter( of_class ),
+                      nex::is::classname(classname) );
+
+        std::vector< std::string > vec;
+        vec.reserve( of_class.size() );
+        std::transform( of_class.begin(), of_class.end(),
+                        std::back_inserter( vec ),
+                        nex::get::varname_str);
+        auto last = std::unique(vec.begin(), vec.end());
+        std::sort(vec.begin(), last);
+        last = std::unique(vec.begin(), last);
+        vec.erase(last, vec.end());
+        return vec;
+    }
+};
+
+NexusPlot load( const std::string& );
+NexusPlot load( std::istream& );
```

Comment:
I do not see the `istream` overload as being relevant - please remove it.

Reply:
The istream overload is used for tests. I could be removed but it makes testing the exceptions easier. The alternative is to just create additional test data that hold the same information as the streams.